diesel_cli/src/main.rs

```diff
@@ -186,6 +188,59 @@ fn convert_absolute_path_to_relative(target_path: &Path, mut current_path: &Path
     result.join(target_path.strip_prefix(current_path).unwrap())
 }
 
+fn run_infer_schema(matches: &ArgMatches) {
+    let database_url = database::database_url(matches);
+    let schema_name = None;
+    let schema = diesel_infer_schema::infer_schema_for_schema_name(&database_url, schema_name);
```

Comment:
I've implement the error managment in a other way to have all infer_schema related code in the corresponding crate. Otherwise the wrapping with the schema module would be done seperatly in codegen and cli. So everything is shared between codegen and cli. The only difference is what happens in case of an error. Codegen panics, cli will print a message and then tries to infer the other tables.